src/App.tsx

```diff
@@ -1,39 +1,61 @@
-import React from 'react';
+import { Component } from 'react';
 import './App.scss';
+import { Clock } from './component';
+
+interface State {
+  hasClock: boolean,
+  clockName: string,
+}
 
 function getRandomName(): string {
   const value = Date.now().toString().slice(-4);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+export class App extends Component<{}, State> {
+  state: State = {
+    hasClock: true,
+    clockName: 'Clock-0',
+  };
+
+  timerId = 0;
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+  componentDidMount() {
+    document.addEventListener('contextmenu', this.handleRightClick);
+    document.addEventListener('click', this.handleLeftClick);
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+    this.timerId = window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
+  }
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+  componentWillUnmount() {
+    document.removeEventListener('contextmenu', this.handleRightClick);
+    document.removeEventListener('click', this.handleLeftClick);
+    if (this.timerId) {
+      window.clearInterval(this.timerId);
+    }
```

Comment:
Also you no need you conditional there
```suggestion
      window.clearInterval(this.timerId);
```